communication/nats/connector.go

```diff
@@ -86,10 +86,10 @@ func (b *BrokerConnector) ReconnectAll() {
 	var wg sync.WaitGroup
 	for k, v := range b.registry {
 		wg.Add(1)
-		go func(id uuid.UUID, conn Connection) {
+		go func(id uuid.UUID, conn *ConnectionWrap) {
 			defer wg.Done()
 			log.Info().Msgf("Re-establishing broker connection %v", id)
-			err := conn.Check()
+			err := conn.Conn.Flush()
```

Comment:
`Check` does `Flush` inside with timeout. How these changes should resolve the problem?

Reply:
Check does flush with 3 seconds timeout. Conn.Flush does it with 60 seconds timeout. I tested this branch on my local pc and it seems work except that now I need to wait much longer to disconnect.

Original issue suggests that session destroy should be sent before VPN tunnel disconnect., but these changes doesn't address that.

Reply:
There is a case, when we are connected to a node, but node does not forward traffic. In such case reordering session-destroy with actual vpn disconnect does not help. 

My suggestion is to implement session-destroy send with retry and attempt to resend up to 60sec. Since we DO get error on failed send.

Reply:
@zolia Why do we forward broker traffic via VPN? Maybe we could forward broker traffic via default consumer gateway?

Reply:
> My suggestion is to implement session-destroy send with retry and attempt to resend up to 60sec. Since we DO get error on failed send.

I tested with retry, but every message sending could end up being slow then.
Left this approach, because:
- no point to do ReconnectAll() wo checking if it works after recconnect
- it's not slow, it takes as much time, as user gets his internet back
- slow parts are postponed to be done after actual reconnect, non blocking anything

Reply:
After digging why it's slow found that `Flush()` does no reconnect, but rather sends "ping" message and waits go NAT reconnect routine job to kick in. So it "ping" should be send with timeout > 5s (see `nats.PingInterval`).

Anyway, it's a hacky way, so remade it to do real reconnect.